### DISCUSSION TITLE
minecraft acc login help

### DISCUSSION THREAD

**Discussion by parv-lj** [PrismarineJS/mineflayer#3339] [General]
hi i logged in microsoft acc with code method.  which means code gave me url i went to that and logged in. what my problem is now i am not able to remove that acc. even though i changed password of my microsoft account it still tries to login into same account in my code. i want help how to remove that token which logs into my account so that next time it asks for new login and gives me the login code.
my program code
const mineflayer = require('mineflayer');

// Function to perform random actions
function performRandomAction(bot) {
    // Define an array of possible actions
    const actions = ['forward', 'back', 'left', 'right', 'jump'];

    // Choose a random action from the array
    const randomAction = actions[Math.floor(Math.random() * actions.length)];

    // Perform the chosen action
    bot.setControlState(randomAction, true);

    // After a random duration, stop the action
    const duration = Math.floor(Math.random() * 3000) + 1000; // Random duration between 1 and 4 seconds
    setTimeout(() => {
        bot.clearControlStates();
    }, duration);
}

// Create the bot
const bot = mineflayer.createBot({
    host: 'kkkkkkk', // minecraft server ip

    port: 25165,
    auth: 'microsoft' // for offline mode servers, you can set this to 'offline'
  
    // version: false,           // only set if you need a specific version or snapshot (ie: "1.8.9" or "1.16.5"), otherwise it's set automatically
    // password: '12345678'      // set if you want to use password-based auth (may be unreliable). If specified, the `username` must be an email
});

// Event handler for when the bot is ready
bot.once('spawn', () => {
    console.log('Bot spawned and ready to perform actions randomly.');

    // Set an interval to perform random actions every 5 seconds
    setInterval(() => {
        performRandomAction(bot);
    }, 5000);
});

// Event handler for errors
bot.on('error', (err) => {
    console.error('Bot encountered an error:', err);
});
bot.end();

**Comment by Givonar** [comment]
in README say :
If `auth` is set to `microsoft`, you will be prompted to login to microsoft.com with a code in your browser. After signing in on your browser, 
the bot will automatically obtain and cache authentication tokens (under your specified username) so you don't have to sign-in again. 

To switch the account, update the supplied `username`. By default, cached tokens will be stored in your user's .minecraft folder, or if `profilesFolder` is specified, they'll instead be stored there.
For more information on bot options see node-minecraft-protocol's [API doc](https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/docs/API.md#mccreateclientoptions).

**Reply by Parvshah-01** [reply]
wow i didnt understand why i got a mail. lol  1.5 years old question got answered.
thanks for the answer though. i dont remember what workaround i did that time. 
cheers
edit : parv-lj is my alt acc
